src/basic/random-util.c

```diff
@@ -61,7 +61,7 @@ int acquire_random_bytes(void *p, size_t n, bool high_quality_required) {
                 r = getrandom(p, n, GRND_NONBLOCK);
                 if (r > 0) {
                         have_syscall = true;
-                        if (r == n)
+                        if (r == (int) n)
```

Comment:
we should always cast the smaller type to the larger, not the larger to the smaller really... Hence, this should really be:

```c
if ((size_t) r == n)
        …
```

Not that this matters IRL, but I think it's a matter of cleanliness...